### PR TITLE
Make sure deeper learning never uses published states of pilot, preview, and stable

### PR DESCRIPTION
**Pull request by dmcavoy**
Deeper learning courses have a separate way they are launched. Their published states should always be 'in_development' or 'beta. Otherwise they will show up in the assignment dropdown. This PR enforces that through a validation. 

## Links

[Jira](https://codedotorg.atlassian.net/browse/PLAT-1724)

## Testing story

- Added/Updated unit tests

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -158,6 +158,13 @@ def hide_pilot_units
     }
 
   validates :published_state, acceptance: {accept: SharedCourseConstants::PUBLISHED_STATE.to_h.values.push(nil), message: 'must be nil, in_development, pilot, beta, preview or stable'}
+  validate :deeper_learning_courses_cannot_be_launched
+
+  def deeper_learning_courses_cannot_be_launched
+    if old_professional_learning_course? && (launched? || pilot?)
+      errors.add(:published_state, 'must be in_development or beta for a deeper learning course.')
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb]: there are more states than this now, and I'm not sure what would happen if we ended up in one of the other ones: https://github.com/code-dot-org/code-dot-org/blob/5ee5b2ce43ce68ff9a52472ab281023dd69716be/lib/cdo/shared_constants/curriculum/shared_course_constants.rb#L10-L12

can you make this more specific to require in development or beta?

**Review by davidsbailey — APPROVED** [review]
LGTM after addressing comment.

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/models/script.rb]: We actually are using the deprecated state right now for a bunch of deeper learning courses. Example: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/scripts_json/csp1-dlp-18.script_json#L15 . So I think what I need to update is the comments and error message. I will do that now

Comment (davidsbailey) [reply dashboard/app/models/script.rb]: Interesting. Are deprecated courses visible in the dropdowns?

Comment (dmcavoy) [reply dashboard/app/models/script.rb]: Only pilot or launched courses will show for uses in the dropdown https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/concerns/curriculum/assignable_course.rb. A levelbuilder or someone with an editor experiment would be able to see them in the dropdown. I was planning to filter them out in the future but wanted to make sure product had thought through everything first

Comment (dmcavoy) [reply dashboard/app/models/script.rb]: https://codedotorg.atlassian.net/browse/PLAT-1790

**Pull request merged**